File: NextAdmin.Core.API/ViewModels/Args/EntityArgs.cs
namespace NextAdmin.Core.API.ViewModels.Args
{
    public class EntityArgs
    {
        public string? EntityName { get; set; }

        public string? CustomActionName { get; set; }

        public Dictionary<string, object>? CustomActionArgs { get; set; }

        public Dictionary<string, object>? Parameters { get; set; }

    }
}

File: NextAdmin.Core.Front/Src/Models/EntityArgs.ts
namespace NextAdmin.Models {

    export interface EntityArgs {

        entityName: string;

        customActionName?: string;

        customActionArgs?: {};

        parameters?: Record<string, any>;

    }



    export interface GetEntitiesArgs extends EntityArgs, Query {

        whereQuery?: string;

        whereQueryArgs?: Array<any>;

        selectQueries?: Array<string>;

        isSelectDistinctQuery?: boolean;

        orderByQueries?: Array<any>;

        skipRecordCount?: number;

        takeRecordCount?: number;

    }


    export interface GetEntityArgs extends EntityArgs {

        entityId?;

        lockKey?: string;

        detailToLoadNames?: Array<string>;
    }


    export interface SaveEntityArgs extends EntityArgs {

        entity: any;

        lockKey?: string;

        conflictAction?: ConflictAction;

    }

    export interface SaveEntitiesArgs extends EntityArgs {

        entitiesToAddOrUpdate?: Array<any>;

        entitiesToDeleteIds?: Array<any>;
    }

    export enum ConflictAction {
        overwrite,
        cancel
    }

}